client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx

```diff
@@ -7,35 +8,36 @@ class RelatedItemsAndOutfit extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      currProdId: '40344',
       currProd: [],
       prodsInfo: [],
       prodsStyles: [],
       prodsMeta: [],
       isFetching: true,
     };
-    this.onRelatedCardClick = this.onRelatedCardClick.bind(this);
   }
 
   componentDidMount() {
-    const { currProdId } = this.state;
-    this.getCurrProdData(currProdId);
-    this.getRelatedData(currProdId);
+    this.setCurrProdToState();
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { currProdId } = this.state;
-    if (prevState.currProdId !== currProdId) {
-      this.getCurrProdData(currProdId);
-      this.getRelatedData(currProdId);
+  componentDidUpdate(prevProps) {
+    const {
+      currProdId,
+    } = this.props;
+    /* console.log('doesnt fire request on "update" --
+    why is the method reaching here on initial page load?'); */
+    if ((prevProps.currProdId !== currProdId)) {
+      this.setCurrProdToState(currProdId);
+      // console.log('it ran once per click!');
     }
   }
 
-  onRelatedCardClick(productId) {
-    // console.log(productId);
-    this.setState({
-      currProdId: productId,
-    });
+  setCurrProdToState() {
+    const {
+      currProdId, prodInfo, prodStyles, prodReviewsMeta,
+    } = this.props;
+    this.setState({ currProd: [prodInfo, prodStyles, prodReviewsMeta] });
+    this.getRelatedData(currProdId);
```

Comment:
I recommend putting this line as a callback to this.setState, setState takes an optional CB to fire immediately after the states are set. https://stackoverflow.com/questions/42038590/when-to-use-react-setstate-callback

Reply:
done! thanks